mtags/src/main/scala-3/scala/meta/internal/pc/completions/NamedArgCompletions.scala

```diff
@@ -104,12 +116,70 @@ object NamedArgCompletions:
     val params: List[Symbol] =
       allParams.filter(param => param.name.startsWith(prefix))
 
+    val completionSymbols = indexedContext.scopeSymbols
+    def matchingTypesInScope(paramType: Type): List[String] =
+      completionSymbols
+        .collect {
+          case sym
+              if sym.info <:< paramType && sym.isTerm && !sym.info.isNullType && !sym.info.isNothingType && !sym
+                .is(Flags.Method) =>
+            sym.decodedName
+        }
+        .filter(name => name != "Nil" && name != "None")
+        .sorted
+
+    def findDefaultValue(param: Symbol): String =
+      val matchingType = matchingTypesInScope(param.info)
+      if matchingType.size == 1 then s":${matchingType.head}"
+      else if matchingType.size > 1 then s"|???,${matchingType.mkString(",")}|"
+      else ":???"
+
+    def fillAllFields(): List[CompletionValue] =
+      val suffix = "autofill"
+      val shouldShow =
+        allParams.exists(param => param.name.startsWith(prefix))
+      val isExplicitlyCalled = suffix.startsWith(prefix)
+      val hasParamsToFill = allParams.count(!_.is(Flags.HasDefault)) > 1
+      if (shouldShow || isExplicitlyCalled) && hasParamsToFill && clientSupportsSnippets
+      then
+        val editText = allParams.zipWithIndex
+          .collect {
+            case (param, index) if !param.is(Flags.HasDefault) => {
+              s"${param.nameBackticked.replace("$", "$$")} = $${${index + 1}${findDefaultValue(param)}}"
+            }
+          }
+          .mkString(", ")
+        List(
+          CompletionValue.namedArg(
+            label = "Autofill with default values",
+            allParams.head,
+            Some(editText),
+          )
+        )
+      else List.empty
+      end if
+    end fillAllFields
+
+    def findPossibleDefaults(): List[CompletionValue] =
+      params.flatMap { param =>
+        val allMembers = matchingTypesInScope(param.info)
+        allMembers.map { memberName =>
+          val editText =
+            param.nameBackticked + " = " + memberName + " "
+          CompletionValue.namedArg(
+            label = editText,
+            param,
+            Some(editText),
```

Comment:
What do you think about adding `defaultValue: Option[String]` to `CompletionValue.NamedArg` and construct the insertText on `CompletionsProvider` site? (Instead of adding `insertText: Option[String]` to `NamedArgument`, and construct it here).

Sorry for the mess, but I want to keep the `insertText` construction as a `CompletionsProvider`'s responsibility. (So that we can easily find where the `insertText` is constructed).


(TBH, I don't like having `insertText` field to `CompletionValue`, though it was needed for `InterpolationCompletion` but it should be limited to `InterpolationCompletion`)

Reply:
Maybe we can remove `insertText` from here altogether, because it is the same as label?